tensorflow_datasets/video/moving_sequence.py

```diff
@@ -0,0 +1,419 @@
+  """
+  x0 = tf.convert_to_tensor(x0)
+  velocity = tf.convert_to_tensor(velocity)
+  if x0.shape.ndims != 2:
+    raise ValueError("x0 must be a rank 2 tensor")
+  if velocity.shape.ndims != 2:
+    raise ValueError("velocity must be a rank 2 tensor")
+  if t.shape.ndims != 1:
+    raise ValueError("t must be a rank 1 tensor")
+  x0 = tf.expand_dims(x0, axis=1)
+  velocity = tf.expand_dims(velocity, axis=1)
+  dx = velocity * tf.expand_dims(tf.expand_dims(t, axis=0), axis=-1)
+  linear_trajectories = x0 + dx
+  assert linear_trajectories.shape.ndims == 3, \
+    "linear_trajectories should be a rank 3 tensor"
+  return linear_trajectories
+
+
+def bounce_to_bbox(points):
+  """
+  Bounce potentially unbounded points to [0, 1].
+
+  Bouncing occurs by exact reflection, i.e. a pre-bound point at 1.1 is moved
+  to 0.9, -0.2 -> 0.2. This theoretically can occur multiple times, e.g.
+  2.3 -> -0.7 -> 0.3
+
+  Implementation
+  points <- points % 2
+  return min(2 - points, points)
+
+  Args:
+    points: float array
+
+  Returns:
+    tensor with same shape/dtype but values in [0, 1].
+  """
+  points = points % 2
+  return tf.math.minimum(2 - points, points)
+
+
+MovingSequence = collections.namedtuple(
+  "MovingSequence",
+  ["image_sequence", "trajectories", "start_positions", "velocities"])
+
+
+def images_to_moving_sequence(
+    images, sequence_length=20, speed=0.1, total_padding=(36, 36),
+    **kwargs):
+  """
+  Convert images to a moving sequence.
+
+  Args:
+    images: [?, in_h, in_w, n_channels] uint8 tensor of images.
+    sequence_length: int, length of sequence.
+    speed: float, length of each step. Scalar, or rank 1 tensor with length
+        the same as images.shape[0].
+    total_padding: (pad_y, pad_x) total padding to be applied in each dimension.
+    kwargs: passed to `create_merged_moving_sequence`
+
+  Returns:
+    `MovingSequence` namedtuple containing:
+        `image_sequence`:
+            [sequence_length, in_h + pad_y, in_w + pad_x, n_channels] uint8.
+        `trajectories`: [sequence_length, n_images, 2] float32 in [0, 1].
+        `start_positions`: [n_images, 2] float32 initial positions in [0, 1].
+        `velocities`: [n_images, 2] float32 normalized velocities.
+  """
+  images = tf.convert_to_tensor(images, dtype=tf.uint8)
+  total_padding = tf.TensorShape(total_padding)
+  speed = tf.convert_to_tensor(speed, dtype=tf.float32)
+  n_images = images.shape[0].value
+  trajectories, x0, velocity = get_random_trajectories(
+    n_images, sequence_length, ndims=2, speed=speed,
+    dtype=tf.float32)
+  sequence_pad_lefts = tf.cast(
+    trajectories * tf.cast(total_padding, tf.float32),
+    tf.int32)
+  sequence = create_merged_moving_sequence(
+    images, sequence_pad_lefts, total_padding, **kwargs)
+  return MovingSequence(
+    image_sequence=sequence,
+    trajectories=trajectories,
+    start_positions=x0,
+    velocities=velocity)
+
+
+def as_moving_sequence_dataset(
```

Comment:
Let’s rm this function and instead have users directly use `images_to_moving_sequence` in their own map functions.